src/main/java/com/kodilla/ecommercee/service/CartService.java

```diff
@@ -35,6 +39,26 @@ public Cart createCart() {
         return cartRepository.save(new Cart());
     }
 
+    public Cart saveCart(Cart cart) {
+        return cartRepository.save(cart);
+    }
+
+    public Cart getCart(Long cartId) throws CartNotFoundException {
+        if(cartRepository.getOne(cartId) == null ){
+            throw new CartNotFoundException();
```

Comment:
Druga sprawa to rodzaje błędu. Michał miał wspomnieć, przygotowałem taki błąd globalny exception EntityNotFoundException("tekst"), który wyrzuca po stronie frontu error=404 not found. Może rozważyłabyś zastosowanie jego wszedzie tam, gdzie nie mozna czegoś znaleźć w bazie (item, cart, order, product, group, user - jeden wspólny exception). Pozbędziesz się też throws w naglówkach metod. Jak to działa pod linkiem https://bykowski.pl/obsluga-wyjatkow-w-rest-api-jak-zrobic-to-elegancko-w-spring-boot/

Reply:
Wstawiłam ten błąd bo ciekawe rozwiązanie! nie zmieniałam na razie order controllera zeby zametu nie wprowadzac ale docelowo tam tez bym wprowadzila te zmiany.